### PR TITLE
Shiny Tab Reload

### PR DESCRIPTION
**Pull request by emichaud998**
Changes in this PR:
The shiny dashboard URL now contains a search parameter containing information on the tab it is currently on, which looks like <shinyURL>?tab=<current tab>. The tab that 'tab' in the url is equal to should switch to the current tab on every tab switch. Now, when the dashboard is reloaded the shiny dashboard can now check to see what the last tab the dashboard was on in the url search parameter, and then it will redirect the user to that tab when the dashboard reloads. This also allows a url that directs to a specific dashboard tab to be shared and opened in a new window.

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: [Lantern 228 JIRA Ticket](https://oncprojectracking.healthit.gov/support/browse/LANTERN-228)
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [ ] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [ ] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

File: shinydashboard/lantern/server.R [hunk 1]
```diff
@@ -1,5 +1,19 @@
 # Define server function
 function(input, output, session) {
+  observeEvent(session, {
```
Comment (bmath10) [inline shinydashboard/lantern/server.R:3]: is this just called once on page load? i'm just trying to understand the difference between these two observeEvents; if you could add a comment that would be super helpful!

**Comment by bmath10** [comment]
I don't know if this is worthwhile to address, but i noticed if I add a nonsense string equal to the tab value in the url, it directs to the main dashboard page (which makes sense) but then keeps the value in the url instead of switching it to "dashboard_tab". I just figured I'd mention it and see what your thoughts were!

Another thing I noticed, which is not directly related to this ticket, is that adding nonsense to the url in general (like before the ?tab) directs to a generic "Not Found" page. I would expect it to direct back to the main page or a Not Found page that we've created. I don't think this is particularly pressing, but I wanted to note it as well since I came across it!

File: shinydashboard/lantern/server.R (hunk 1 above)
Comment (emichaud998) [reply shinydashboard/lantern/server.R:5]: I added some comments to the observeEvent functions to make this more clear!

**Comment by emichaud998** [comment]
> I don't know if this is worthwhile to address, but i noticed if I add a nonsense string equal to the tab value in the url, it directs to the main dashboard page (which makes sense) but then keeps the value in the url instead of switching it to "dashboard_tab". I just figured I'd mention it and see what your thoughts were!
> 
> Another thing I noticed, which is not directly related to this ticket, is that adding nonsense to the url in general (like before the ?tab) directs to a generic "Not Found" page. I would expect it to direct back to the main page or a Not Found page that we've created. I don't think this is particularly pressing, but I wanted to note it as well since I came across it!

Good point Bri! In terms of the nonsense string problem, I just fixed this so now, if you add a nonsense string equal to the tab value in the url, it not only directs to the main dashboard page but then it switches the value to "dashboard_tab" in the url! In terms of the Not Found page, I am not entirely sure what might be causing this as this was not a part of this ticket as you mentioned, but if it is something I should look into for this PR, let me know!

**Comment by bmath10** [comment]
> Good point Bri! In terms of the nonsense string problem, I just fixed this so now, if you add a nonsense string equal to the tab value in the url, it not only directs to the main dashboard page but then it switches the value to "dashboard_tab" in the url! In terms of the Not Found page, I am not entirely sure what might be causing this as this was not a part of this ticket as you mentioned, but if it is something I should look into for this PR, let me know!

It looks good! I don't think the Not Found page issue has anything to do with this PR, so I'll just add a ticket to the backlog!

**Review by bmath10 — APPROVED** [review]

**Review by mayerm94 — APPROVED** [review]

**Pull request merged**